x-pack/test/api_integration/apis/entity_store/host.ts

```diff
@@ -0,0 +1,191 @@
+          expect(engine.status).to.eql('started');
+          for (const component of engine.components) {
+            expect(component.installed).to.be(true);
+          }
+        }
+      });
+
+      it('Should successfully trigger a host transform', async () => {
+        const HOST_NAME: string = 'host-transform-test-ip'
+        const IPs: string[] = ['1.1.1.1', '2.2.2.2']
+        let response = await es.transform.getTransformStats({
+          transform_id: HOST_TRANSFORM_ID,
+        });
+        expect(response.count).to.eql(1);
+        let transform = response.transforms[0];
+        expect(transform.id).to.eql(HOST_TRANSFORM_ID);
+        const triggerCount: number = transform.stats.trigger_count;
+        const docsProcessed: number = transform.stats.documents_processed;
+
+        // Create two documents with the same host.name, different IPs
+        for (let ip of IPs) {
+          const { result } = await es.index(
+            buildHostTransformDocument(HOST_NAME, {ip: ip})
+          );
+          expect(result).to.eql('created');
+        }
+
+        // Trigger the transform manually
+        const { acknowledged } = await es.transform.scheduleNowTransform({
+          transform_id: HOST_TRANSFORM_ID,
+        });
+        expect(acknowledged).to.be(true);
+
+        await retry.waitForWithTimeout('Transform to run again', TIMEOUT_MS, async () => {
+          let response = await es.transform.getTransformStats({
+            transform_id: HOST_TRANSFORM_ID,
+          });
+          let transform = response.transforms[0];
+          expect(transform.stats.trigger_count).to.greaterThan(triggerCount);
+          expect(transform.stats.documents_processed).to.greaterThan(docsProcessed);
+          return true;
+        });
+
+        await retry.waitForWithTimeout('Document to be processed and transformed', TIMEOUT_MS, async () => {
+          const result = await es.search({
+            index: INDEX_NAME,
+            query: {
+              term: {
+                "host.name": HOST_NAME,
+              },
+            },
+          })
+          expect(result.hits.total.value).to.eql(1);
+          expect(result.hits.hits[0]._source.host.name).to.eql(HOST_NAME);
+          expect(result.hits.hits[0]._source.host.ip).to.eql(IPs);
+
+          return true;
+        })
+
+      });
+    });
+  });
+}
+
+type hostObject = {
+  domain: string
+  hostname: string
+  id: string
+  os: {
+    name: string
+    type: string
+  },
+  ip: string
+  mac: string
+  type: string
+  architecture: string
+}
+
+function buildHostTransformDocument(name: string, host: hostObject): Object {
+  // Get timestamp without the millisecond part
+  const isoTimestamp: string = (new Date).toISOString().split('.')[0];
+  let document: Object = {
+    index: DATASTREAM_NAME,
+    document: {
+      '@timestamp': isoTimestamp,
+      host: host,
+    },
+  }
+  document.document.host.name = name;
+  return document
+}
```

Comment:
Comment: Again, is there a ready-to-use Document type? How do I discover stuff like this?

Reply:
you can check the function signature for `es.index` to see what it expects. i think it's `IndexRequest` from `@elasticsearch` package

Reply:
Checked, it's `Object`: https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/api-reference.html#_arguments_18